arch/arm/mach-ctr/bottom_lcd.c

```diff
@@ -92,12 +92,13 @@ void nintendo3ds_bottom_lcd_clear_screen(unsigned int color)
 void nintendo3ds_bottom_lcd_draw_char(const struct font_desc *font, int x, int y, unsigned int color, char c)
 {
 	int i, j;
-	const u8 *src;
+	const u16 *src;
+	const u8 bytes_per_char = 36;//font->charcount / sizeof font->data; // (doesn't work)
```

Comment:
`struct font_desc::data` is a pointer so `sizeof` will return 4.
I think the way to calculate `bytes_per_char` is:
```c
bytes_per_char = ((font->width + 7) / 8) * font->height;
```
Check https://elixir.bootlin.com/linux/latest/source/lib/fonts/font_10x18.c#L5135

But I think this is note needed. Also, I'd prefer if the code was generic for any font size.